src/persist-txn/src/txn_cache.rs

```diff
@@ -327,6 +337,21 @@ impl<T: Timestamp + Lattice + TotalOrder + StepForward + Codec64> TxnsCacheState
         self.compact_data_times()
     }
 
+    pub(crate) fn push_entries(&mut self, mut entries: Vec<(TxnsEntry, T, i64)>, progress: T) {
```

Comment:
This is mostly duplicated from `TxnsCache`, but not quite, because of how `progress_exclusive` is set at the end here. I imagine you don't want to put in the effort to unify that code? (which I'm onboard with!)

Reply:
The first time I tried this while working on this PR, I hit some issues with having to plumb Send+Sync bounds everywhere, but I just tried it again and it worked! \o/ /shrug